nemo/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -92,6 +94,9 @@ def __init__(
         self.deferred_ckpts_to_remove: List[List[str]] = []
         self.ckpts_to_link: Dict[str, str] = {}
 
+        # used to drop optimizer states
+        self.base_checkpoint_io = None
```

Comment:
Is there any reason to keep it as an attribute?
If it's just to avoid recomputing the checkpoint_io unwrapping, I don't think we should store it.